FitGoal/FitGoal/UserProfileConfiguration/ViewControllers/UserProfileConfiguratorViewController.swift

```diff
@@ -14,16 +14,33 @@ class UserProfileConfiguratorViewController: UIViewController {
     private let questionPrefix = UILabel()
     private let questionSuffix = UILabel()
     
-    private let userInfoEntryViewDelegate: UserInfoEntryViewDelegate
+    private var iconListViewDelegate: IconListDelegate?
+    private var userInfoTextFieldDelegate: UserInfoTextFieldDelegate?
     
-    init(userInfoEntryView: UIView, questionPrefix: String, questionSuffix: String, userProfileType: UserProfileType) {
+    private init(userInfoEntryView: UIView, questionPrefix: String, questionSuffix: String) {
         self.userInfoEntryView = userInfoEntryView
-        self.userInfoEntryViewDelegate = UserInfoEntryViewDelegate(userInfoEntryView: userInfoEntryView, userProfileType: userProfileType)
-        
         super.init(nibName: nil, bundle: nil)
         configureQuestions(prefix: questionPrefix, suffix: questionSuffix)
     }
     
+    convenience init(iconListView: IconListView, questionPrefix: String, questionSuffix: String) {
+        
+        self.init(userInfoEntryView: iconListView, questionPrefix: questionPrefix, questionSuffix: questionSuffix)
+        
+        guard let iconListView = userInfoEntryView as? IconListView else { return }
```

Comment:
You don't need this line. Instead, you can use the `iconListView` that is passed as a parameter to the initialiser. This works because `UIView` is a reference type, so both `iconListView` and `userInfoEntryView` will point to the same object.